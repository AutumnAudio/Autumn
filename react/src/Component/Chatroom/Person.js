import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import PersonIcon from '@material-ui/icons/Person';
import AddIcon from '@material-ui/icons/Add';

const styles = {
    personIcon: {
        position: 'relative',
        marginLeft: '10%',
        float: 'left',
        fontSize: '40px'
    },
    nameDiv: {
        positive: 'relative',
        marginLeft: '25%',
        fontSize: '20px',
    },
    currentSongDiv: {
        fontSize: '12px',
        position: 'relative',
        marginLeft: '25%',
        color: '#D0D1D7',
    },
    personDiv: {
        color: 'white',
        margin: '20px',
    },
    addIcon: {
        position: 'absolute',
        marginLeft: '270px',
        marginTop: '-10px',
    }
};

const Person = (props) => {
    const { classes, person, updatePlayListSongs } = props
    var recentlyPlayedSong = ''
    var recentlyPlayedArtist = ''
    var recentlyPlayedSongs = person['recentlyPlayed']
    if (person['recentlyPlayed'][0] != null) {
        recentlyPlayedSong = person['recentlyPlayed'][0]['name']
        recentlyPlayedArtist = person['recentlyPlayed'][0]['artists'][0]
    }
    return (
        <div className={classes.personDiv}>
            <PersonIcon className={classes.personIcon}/>
            <div className={classes.nameDiv}>{person.username}</div>
            <AddIcon className={classes.addIcon}/>
            <div className={classes.currentSongDiv} onClick={() => updatePlayListSongs(recentlyPlayedSongs)}>
                {recentlyPlayedSong} - {recentlyPlayedArtist}
            </div>
        </div>
    )
    
}

Person.propTypes = {
    classes: PropTypes.object.isRequired,
    person: PropTypes.object.isRequired,
    updatePlayListSongs: PropTypes.func.isRequired,
};

export default withStyles(styles)(Person)