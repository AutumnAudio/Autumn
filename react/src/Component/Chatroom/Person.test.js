import React from 'react'
import renderer from 'react-test-renderer'
import Person from './Person'

test('should be able to render people', () => {
    const personMock = {username: 'testUser',
                        recentlyPlayed: [{name: 'song name', artists: ['testArtist']}]}
    const component = renderer.create(
        <Person person={personMock} />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
