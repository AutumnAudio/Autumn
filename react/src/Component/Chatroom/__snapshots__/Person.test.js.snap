// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to render people 1`] = `
<div
  className="Person-personDiv-4"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root Person-personIcon-1"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
  <div
    className="Person-nameDiv-2"
  >
    testUser
  </div>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root Person-addIcon-5"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
  <div
    className="Person-currentSongDiv-3"
  >
    song name
     - 
    testArtist
  </div>
</div>
`;
